### PR TITLE
Added BatchNorm

### LINKED ISSUE #81: fatal error: Cannot create ConcatV2. Missing or invalid attribute axis.
Hello, just update to your new version, seems to work but i got this error when loading my pb model.

guard let axisInt32 = axisNode.nodeDef.attr["value"]?.tensor.intVal.first, let axis = LayerSizeAxis.fromTF(index: Int(axisInt32)) else {
                fatalError("Cannot create \(Constants.Ops.Concat). Missing or invalid attribute axis.")
            }

here axistInt32 = 3 , but cannot create concatv2 when axis = LayerSizeAxis.fromTF(index: Int(axisInt32))
Any idea?
I am basically trying to convert my pb model which is one derived from https://github.com/Joker316701882/Salience-Object-Detection
and then froze. 
Also some warning 
"Unknown input: pool4_1:1"
"Unknown input: pool4:1"
"Unknown input: pool3:1"
"Unknown input: pool2:1"
"Unknown input: pool1:1"
thanks a lot

### PR DESCRIPTION
**Pull request by mats-claassen**
Also added a lot of ops that will be ignored and removed from the graph with their inputs and outputs rewired.

Also added some layers like the old "Concat" (not ConcatV2) which are mapped to similar layers that already exist.

Lastly, concat axis parsing was fixed. Fixes #81

### CODE REVIEW

**Comment by bryant1410** [comment]
Devil bug around!

let the game begin

File: Example/BatchNormTest.swift [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+import AVFoundation
+import MetalBender
+import MetalKit
+import MetalPerformanceShaders
+import MetalPerformanceShadersProxy
+
+struct BatchNormDataSet {
+
+    static let epsilon: Float = 0.001
+    static let mean = [Float].init(repeating: 0.5, count: 8)
```
Comment (dernster) [inline Example/BatchNormTest.swift]: Maybe it's a personal taste but I prefer not writing the `init` keyword here since we almost never write it explicitly.

File: Example/BatchNormTest.swift [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        test2(useScale: false, depth: 4),
+    ]
+
+    static func test1(useScale: Bool, depth: Int = 8) -> Test {
+        let texture1 = Texture(
+            data: [
+                [Float].init(repeating: 1, count: depth), [Float].init(repeating: 2, count: depth),
+                [Float].init(repeating: 3, count: depth), [Float].init(repeating: 4, count: depth),
+                [Float].init(repeating: 5, count: depth), [Float].init(repeating: 6, count: depth),
+                ],
```
Comment (dernster) [inline Example/BatchNormTest.swift]: dedent

File: Example/BatchNormTest.swift [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    }
+
+    static func test2(useScale: Bool, depth: Int = 8) -> Test {
+        var data = [[Float]]()
+        for _ in 0..<3 {
+            for _ in 0..<3 {
+                data.append([Float].random(count: depth))
+            }
+        }
+        let texture1 = Texture(data: data, size: LayerSize(h: 3, w: 3, f: depth))
```
Comment (dernster) [inline Example/BatchNormTest.swift]: texture1 -> texture ?

File: Example/BatchNormTest.swift [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+
+    static let testData: [Test] = [
+        test1(useScale: true, depth: 8),
+        test1(useScale: false, depth: 8),
+        test2(useScale: true, depth: 4),
+        test2(useScale: false, depth: 4),
+    ]
+
+    static func test1(useScale: Bool, depth: Int = 8) -> Test {
+        let texture1 = Texture(
```
Comment (dernster) [inline Example/BatchNormTest.swift]: texture1 -> texture ?

File: Example/BatchNormTest.swift [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+                          epsilon: BatchNormDataSet.epsilon)
+
+        styleNet.initialize()
+        let metalTexture = inputTexture.metalTexture(with: Device.shared)
+        styleNet.run(input: MPSImage(texture: metalTexture, featureChannels: inputTexture.depth)) { image in
+            let textureFromGpu = Texture(metalTexture: image.texture, size: expectedOutput.size)
+            if !textureFromGpu.isEqual(to: expectedOutput, threshold: 0.01) {
+                print("TEST failed:")
+                print(textureFromGpu)
+                print(expectedOutput)
```
Comment (dernster) [inline Example/BatchNormTest.swift]: what about using an `assert` here in order to explicitly fail

File: Example/Example/Tests/Helpers/Float.swift [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+import Foundation
+
+extension Array where Element == Float {
+
+    static func random(count: Int) -> [Float] {
+        var arr = [Float]()
+        for _ in 0..<count {
+            arr.append(Float(arc4random()) / Float(UINT32_MAX))
+        }
+        return arr
```
Comment (dernster) [inline Example/Example/Tests/Helpers/Float.swift]: what about `return (0..<count).map { _ in Float(arc4random()) / Float(UINT32_MAX) }` ?

File: Sources/Metal/batch_norm.metal [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+kernel void batch_norm(texture2d_array<float, access::read> inTexture [[texture(0)]],
+                       texture2d_array<float, access::write> outTexture [[texture(1)]],
+                       constant float4* params[[buffer(0)]],
+
+                       ushort3 gid [[thread_position_in_grid]]
+                       ) {
+
+    float4 i = inTexture.read(ushort2(gid.x, gid.y), gid.z);
+    int depth = inTexture.get_array_size();
+    float4 out = params[2 * depth + gid.z] * (i - params[gid.z]) / params[depth + gid.z] + params[3 * depth + gid.z];
```
Comment (dernster) [inline Sources/Metal/batch_norm.metal]: maybe declare const variables for indexes like `scaleIndex`, `offsetIndex` and so on. Just for readability

File: Sources/Helpers/Helpers.swift [hunk 8]
```diff
@@ -57,3 +57,14 @@ func HWIOtoOWHI(weights: Data, shape: Shape) -> Data {
...
     return Data.init(bytes: transposed, count: shape.totalCount * MemoryLayout<Float>.stride)
 }
+
+public func float32to16(_ input: UnsafeMutablePointer<Float>, count: Int) -> [UInt16] {
+    var output = [UInt16](repeating: 0, count: count)
+    var bufferFloat32 = vImage_Buffer(data: input,   height: 1, width: UInt(count), rowBytes: count * 4)
+    var bufferFloat16 = vImage_Buffer(data: &output, height: 1, width: UInt(count), rowBytes: count * 2)
+
+    if vImageConvert_PlanarFtoPlanar16F(&bufferFloat32, &bufferFloat16, 0) != kvImageNoError {
+        print("Error converting float32 to float16")
```
Comment (dernster) [inline Sources/Helpers/Helpers.swift]: Maybe we need to discuss how to handle errors, but I prefer to fail with a noticeable error here. Maybe an `assert`
